### ISSUE TITLE
[Request for Maintenance] CPP tooling in open-telemetry/build-tools

### ISSUE THREAD

**Issue by jsuereth** [open-telemetry/opentelemetry-cpp#2550]
There's a lot of dependabot failures to upgrade the [cpp tooling in build-tools](https://github.com/open-telemetry/build-tools/tree/main/cpp_format_tools) repository.

As such, I'd like to propose the following:

1. We can create a specific `cpp-build-tools` repository.  This would publish a docker image just like the existing build-tools repo.
2. We grant maintainer access of this repository to ALL C++ maintainers (can include -contrib to if needed).
3. The C++ maintainers would then be responsible for keeping tools (like lint, docker-image) up-to-date.

Today - Given the distributed nature of ownership of the `build-tools` repository, it's hard and/or impractical for CPP maintainers to make fixes to build-tools, get reviews and update their projects.

I'm unfortunately unable to attend your SiG due to conflicting meetings, but am happy to discuss the process of passing maintenance of these tools to the C++ maintainers on this issue or (if necessary) in the specification meeting.

Related: https://github.com/open-telemetry/opentelemetry-specification/issues/3899
Related: https://github.com/open-telemetry/semantic-conventions/issues/767

**Comment by esigo** [comment]
I support the idea of new repository with the possibility to publish docker image.

**Comment by marcalff** [comment]
Hi @jsuereth 

I support this idea also.

**Comment by lalitb** [comment]
I think it was earlier discussed and agreed to remove this tooling - https://github.com/open-telemetry/opentelemetry-cpp/issues/2050
cc @bogdandrutu

**Comment by ThomsonTan** [comment]
I support this idea of moving `cpp-build-tools` to a new repo.

**Comment by marcalff** [comment]
> I think it was earlier discussed and agreed to remove this tooling - #2050 cc @bogdandrutu

It is correct the opentelemetry-cpp CI does not use the docker image provided, which is why the decision then was to remove it.

Since then, I had to reinstall a new computer from 0, and discovered how actually useful the docker image is to format all the code, especially for cmake-format, when all the tooling is not available in the distribution used.

For new contributors submitting a PR for the first time, clang-format and cmake-format are a strong barrier.

Keeping the docker image available, assuming we advertise it more, will help.

In short, I am reverting my previous opinion, and now prefer to keep the tooling.

**Comment by lalitb** [comment]
Got it, makes sense to have a separate repo that publishes a docker image of this tooling.

**Comment by jsuereth** [comment]
@marcalff Should I assume you'd be the primary owner then helping me move this tooling into its own repository?

I can set up the repository with all existing CPP maintainers/approvers and help get the docker-image publishing working, then let you take over from there, if this all makes sense.

Accepting this issue (triage/accepted) label will tell me to go forward and I'll start creating the repository.

**Comment by marcalff** [comment]
@jsuereth

Yes, agreed.

**Comment by jsuereth** [comment]
When you're comfortable with new repo, please approve: https://github.com/open-telemetry/build-tools/pull/275

Your new repo should be able to build docker images from main every time you merge a commit.

**Comment by marcalff** [comment]
Thanks @jsuereth 

Is there a way for cpp-maintainers to get admin privileges on this new repo ?

We need to setup branch protection rules, enforce easyCLA, etc.

**Comment by tigrannajaryan** [comment]
I added DOCKER_USERNAME/DOCKER_PASSWORD repository secrets to https://github.com/open-telemetry/cpp-build-tools/settings/secrets/actions

You should be able to use them in the actions.

**Comment by jsuereth** [comment]
@marcalff Branch protection rules and easyCLA should already be set up for you (TC does that when creating any repository for consistency)

Thanks @tigrannajaryan for setting up the secrets!!!

**Comment by esigo** [comment]
no easyCLA check in the repo https://github.com/open-telemetry/cpp-build-tools/pull/4

**Comment by jsuereth** [comment]
EasyCLA is configured at the organization level to be on for all repositories, let me ask around I may ahve missed a check in the repo setup that's preventing it from commenting: https://github.com/open-telemetry/community/blob/main/docs/how-to-configure-new-repository.md

**Comment by marcalff** [comment]
@jsuereth 

The team https://github.com/orgs/open-telemetry/teams/cpp-maintainers/repositories has maintain privilege, this part seems ok.

The team https://github.com/orgs/open-telemetry/teams/cpp-approvers/repositories only has the triage privilege, this should be write instead.

**Comment by jsuereth** [comment]
@marcalff fixed.

**Comment by jsuereth** [comment]
I'm going to close out this issue.  Thank you for taking the new repository!  Please open bugs in `community` or ping me on Slack if you need any help or have any problems going forward!!!

Thank you again!
